WWP_seq_new_run_auto.sh

```diff
@@ -0,0 +1,120 @@
+echo "$(date) : The sample sheet for run $run_name is $sample_sheet"
+
+echo "$(date) : Generating list of wastewater samples from the run sample sheet. Used to fetch matching fastq files in the next step"
+# Though, the sample sheet by default will only contain wastewater samples, doing this to avoid additional files such as 'undetermined_*.fastq.gz' to be included in the downstream analysis 
+grep -i 'Wastewater' $sample_sheet | cut -f 1 -d ',' > $analysis_dir/${run_name}_wastewater_sample_list.csv
+
+# Fastq files directory source
+fastq_dir=$analysis_dir/raw_data
+# Fastq files directory destination
+ww_fastq=$analysis_dir/raw_data/fastq
+
+echo "$(date) : Copying fastq files from $fastq_dir to $ww_fastq directory to run with viralrecon"
+find $fastq_dir -type f -name '*.fastq.gz' -print0 | grep -zf $analysis_dir/${run_name}_wastewater_sample_list.csv| parallel -0 "mv {} $ww_fastq/"
+
+echo "$(date) : Fastq files copied successfully to $ww_fastq"
+cd $ww_fastq
+
+# Samples that fail after the run are excluded from downstream analysis. This step is nececssary for running samples with viralrecon otherwise oftentimes, the pipeline fails
+echo "$(date) : Moving failed samples from raw_data directory to another directory (failed_samples) based on file size < 1MB"
+find $ww_fastq -type f -name "*.fastq.gz" -size -1M -print0 | parallel -0 mv {} $analysis_dir/failed_samples/ 
+
+# Rename fastq files for bioinformatics analysis. Also copy fastq files to NCBI submission directory and rename.
+echo "$(date) : Rename fastq files for downstream analysis and for NCBI submission"
+
+# If there are two files (_L001 and _L002) per sample as in the case of P3 flow cell then this step is needed to merge the fastqs from two lanes
+
+#mkdir unmerged
+#for sample in $(ls *.fastq.gz | sed 's/_L00[0-9]_R1_001.fastq.gz//' | sort | uniq); do 
+#    cat ${sample}_L00*_R1_001.fastq.gz > ${sample}_L001002_R1_001.fastq.gz;
+#    for unmerged in ${sample}_L00*_R1_001.fastq.gz; do
+#        if [ "$unmerged" != "${sample}_L001002_R1_001.fastq.gz" ]; then
+#            mv $unmerged unmerged
+#        fi
+#    done
+#done
+
+for file in *_R1_001.fastq.gz; do
+    # Check if the file contains the run_name (won't be true for NextSeq 2000 runs)
+    if [[ "${file}" == *"${run_name}"* ]]; then
+        # Remove the lane and Set identifiers
+        new_name=$(echo "${file}" | sed -E "s/_S[0-9]+_L[0-9]+//")
+        echo "${new_name}"
+        mv "${file}" "${new_name}"
+    else
+        # Add the run name and remove the lane and Set identifiers
+        new_name=$(echo "${file}" | sed -E "s/_S[0-9]+_L[0-9]+_/-${run_name}_/")
+        echo "${file}"
+        echo "${new_name}"
+        mv "${file}" "${new_name}"
+    fi
+    echo "$(date) : Copy the renamed files to NCBI submission directory"
+    cp "${new_name}" "${analysis_dir}/ncbi_submission/$(echo "${new_name}" | sed -E 's#\-(A01290|VH00770).*##')_R1.fastq.gz" 
+done
```

Comment:
This is a brain freeze on my part, but does this still work when you get multiple files for the same sample?

Reply:
Sorry about that. It looks complicated because it's intended to handle filenames on both NextSeq and NovaSeq (they have different naming schemes for fastqs). Currently, it handles one fastq file per sample. I haven't tested it yet on paired fastq files as we haven't been doing paired sequencing. However, If there are multiple files like _L1 and _L2, I have some commented code on lines 72-82 to handle those use cases. 

I can surely test this on paired fastqs as we might be starting paired sequencing soon.

Reply:
I think it's fine to work with what you're currently seeing. I think it'd be more worthwhile to make adjustments for paired-end files when you start to see them.